### DISCUSSION TITLE
vscode: can`t use github copilot with remote-ssh

### DISCUSSION THREAD

**Discussion by HJDCZY** [community/community#52369] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I use vscode to programming. I always use remote-ssh by microsoft  and github copilot to write codes on my server. 
![image](https://user-images.githubusercontent.com/112754310/230772371-58b5c138-0ce7-4bc4-9ee1-334d0fd8554b.png)

today when I connected my server by openssh, github copilot can not connect to the server
![image](https://user-images.githubusercontent.com/112754310/230772438-253bc0bf-0418-489e-b8c3-ac4708cb466f.png)

and when I click "retry", it  still can not connect ,log like this
![image](https://user-images.githubusercontent.com/112754310/230772470-c7510438-26b4-4c3a-af48-1ba93b68a0c8.png)

I can use github copilot in locality , but can use in remote-ssh of my server. are there any ways to fix?

**Comment by Akash1134** [comment]
Hi @HJDCZY ,

Have you set up your proxy authentication using this VSCode setting?

The problem might also be related to  trust in chain certificates.
1 - install the extension in VSC "win-ca"
2 - change config "Win-ca: inject (Method to provide fetched root certificates to other extensions)" to append.
3 - restart VSC.

Let us know how it goes. Thanks!

**Reply by HJDCZY** [reply]
when I do as you say, copilot would be alright for a while. but when I uses copilot for 3~4 times, it comes error. icon like this:
![image](https://user-images.githubusercontent.com/112754310/232056746-372fc247-7e41-4d57-a7f8-8d67b6584d49.png)
and "output" like this:
![image](https://user-images.githubusercontent.com/112754310/232056940-c560aeec-305f-4379-9625-92d3019dd670.png)

**Reply by Akash1134** [reply]
This error looks like an issue related to hosts file setup (something like /etc/hosts file entries) on your proxy network and is generic to your IDE.

**Reply by HJDCZY** [reply]
I do not have this kind of problem and copilot is normal when programming locally. this problem exists only when I use remote-ssh. I have already deleted hosts file and create an empty one. but the same problem still exists. You say that the problem is caused by proxy, I have already closed all proxy and VPN software but the problem still exist.  Do you have any solution ? thank a lot!

**Comment by HJDCZY** [comment, marked as answer]
okay, I have solved this problem. you can`t use any proxy network in your remote server. I found the problem when I restart the server.

**Reply by piwawa** [reply]
What do you mean? Should i set no `http_proxy`? My server is in CHINA, I can't connect the copilot server without it!

**Reply by Ramilll** [reply]
@piwawa have you solved this issue?

**Comment by amd-vamsit** [comment]
Try this option as well in case if above option didnt worked.
Ref: https://github.com/orgs/community/discussions/6942#discussioncomment-4962858

**Reply by piwawa** [reply]
The setting is useless for me.

**Comment by CasperTsuiPro** [comment]
Thank you so much, adding the following to `setting.json` finally make it works!
```json
"remote.extensionKind": {
  "GitHub.copilot": [
    "ui"
  ],
  "GitHub.copilot-chat": [
    "ui"
  ],
},
```

**Reply by emcmanus** [reply]
Step by step:

`cmd+shift+p` to open the command panel

`>Preferences: Open Remote Settings (JSON) (SSH: ...)`

Update the file to
```
{
  "remote.extensionKind": {
    "GitHub.copilot": [
      "ui"
    ],
    "GitHub.copilot-chat": [
      "ui"
    ],
  },
}
```

Save

`cmd+shift+p` to open the command panel

`>Developer: Reload window`

**Reply by liang-zijian** [reply]
perfect!

**Reply by ihor-drachuk** [reply]
Exactly this answer helped me! Thanks!

**Reply by TTiannaiTT** [reply]
A perfect and easy way! Thanks a lot!

**Reply by bzchen1** [reply]
thank you so much!

**Reply by WarningRoo** [reply]
It's helpful for me! Thank you~

**Reply by nhanledev** [reply]
Adding this to the setting.json file works! Thank a lot!

**Reply by chsivateja** [reply]
You need to add this to local settings.json

**Reply by zhangxc98** [reply]
When I added this to the local settings.json file, it worked, but it made my vscode very clunky, and after canceling it, it returned to normal. How to sove it?

**Reply by ICE0208** [reply]
tysm 🙌

**Reply by MMo-atoss** [reply]
Thanks a lot, finally it works!

**Reply by reducm** [reply]
# Issue: Telemetry Sending Error in Multi - OS Remote SSH Collaboration Scenario

## Description
When editing via remote SSH on an online server in a multi - OS (mac, win, linux) collaborative environment, even after adding the relevant `settings.json` configuration, the following error occurs:

```
[Timestamp] [error] [default] Error sending telemetry FetchError: unable to verify the first certificate
    at fetch ([Path Hidden]/extensions/github.copilot - 1.331.0/node_modules/@adobe/helix - fetch/src/fetch/index.js:99:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at cachingFetch ([Path Hidden]/extensions/github.copilot - 1.331.0/node_modules/@adobe/helix - fetch/src/fetch/index.js:288:16)
    at K1.fetch ([Path Hidden]/extensions/github.copilot - 1.331.0/lib/src/network/helix.ts:94:22) {
  type:'system',
  _name: 'FetchError',
  code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
  errno: undefined,
  erroredSysCall: undefined
}
```

## Environment Information
- Operating Systems Involved: mac, win, linux (Multi - person collaboration scenario, connecting to the online server for editing via remote SSH)
- Relevant Extension: GitHub Copilot (Version 1.331.0)

Hope this telemetry sending error can be resolved to ensure normal operation in the multi - OS collaborative remote SSH scenario.

**Reply by asaarashi** [reply]
what a magic!

**Reply by DziugasMolis** [reply]
This one worked, adding server to /etc/resolv.conf did not help for me

**Reply by infraredZ** [reply]
Copilot cannot operate on files of a remote server; it can only chat.

**Reply by fa1247** [reply]
> Copilot cannot operate on files of a remote server; it can only chat.

Yes, there is a discussion related to this issue. https://github.com/orgs/community/discussions/168848

**Comment by vanarok** [comment]
Copilot doesn't work with remote development extension, use cursor ide!

**Comment by girishlade111** [comment]
you can`t use any proxy network in your remote server.
